src/api/utils/note-header.guard.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import { CanActivate, ExecutionContext, Injectable } from '@nestjs/common';
+
+import { ConsoleLoggerService } from '../../logger/console-logger.service';
+import { Note } from '../../notes/note.entity';
+import { NotesService } from '../../notes/notes.service';
+
+/**
+ * This guard take the note's id in header and find the Note in database.
+ */
+@Injectable()
+export class NoteHeaderGuard implements CanActivate {
+  constructor(
+    private readonly logger: ConsoleLoggerService,
+    private noteService: NotesService,
+  ) {
+    this.logger.setContext(NoteHeaderGuard.name);
+  }
+
+  async canActivate(context: ExecutionContext): Promise<boolean> {
+    const request: Request & {
+      headers: { 'hedgedoc-note': string }; // eslint-disable-line
```

Comment:
I'm not sure whether it makes a difference if the header is written lowercase here, where it was written as `HedgeDoc-Note` previously.

@davidmehren

Reply:
Inside `request.headers` all header names are in lowercase. I don't know if we can use the `@Headers()` function from nest, Its my first time with nest and I didn't find it on documentation.

I have to add this "eslint disable" because eslint doesn't like keys with "-", I try using `hedgedocNote` but it doesn't work.

Reply:
Header names are always lowercased according to the [node docs](https://nodejs.org/api/http.html#messageheaders), so that should be alright.